skimage/draw/draw.py

```diff
@@ -148,28 +149,66 @@ def circle(r, c, radius, shape=None):
     Parameters
     ----------
     r, c : double
-        Centre coordinate of circle.
+        Center coordinate of disk.
     radius : double
-        Radius of circle.
+        Radius of disk.
     shape : tuple, optional
         Image shape which is used to determine the maximum extent of output
-        pixel coordinates. This is useful for circles that exceed the image
-        size. If None, the full extent of the circle is used.  Must be at least
+        pixel coordinates. This is useful for disks that exceed the image
+        size. If None, the full extent of the disk is used.  Must be at least
+        length 2. Only the first two values are used to determine the extent of
+        the input image.
+
+    Returns
+    -------
+    rr, cc : ndarray of int
+        Pixel coordinates of disk.
+        May be used to directly index into an array, e.g.
+        ``img[rr, cc] = 1``.
+
+    Warns
+    -----
+    Deprecated:
+        .. versionadded:: 0.17
+
+            This function is deprecated and will be removed in scikit-image 0.19.
+            Please use the function named ``disk`` instead.
+    """
+    warnings.warn("circle is deprecated in favor of "
+                  "disk."
+                  "circle will be removed in version 0.19",
+                  FutureWarning, stacklevel=2)
+    return disk(r, c, radius, shape=shape)
+
+
+def disk(r, c, radius, *, shape=None):
+    """Generate coordinates of pixels within circle.
+
+    Parameters
+    ----------
+    r, c : double
+        Center coordinate of disk.
```

Comment:
@sciunto since we are making a new function here, how do you feel about starting with the "nD style" API?

```python
def disk(center, radius, *, shape=None):
```

Radius could be an int now but a vector in future versions where we might want to draw disks in an nD space...

On the other hand, maybe it's better to upgrade the whole module in one go, in which case then it's better to use a different PR for that. I'll leave the decision to you!

Reply:
Excellent remark! I think I can change in this PR the signature. Imo, one deprecation is better than two, even if in the mean time it is more heterogeneous. The burden caused by updating existing code is more important.

I'll change the signature asap.

Reply:
@jni can you check https://github.com/scikit-image/scikit-image/pull/4428/commits/070bf872e67d2a1856f46cf0ea60d8b442c229e9 (can you confirm that the order of the center coordinates is correct?